src/room.ts

```diff
@@ -209,21 +230,40 @@ export class Room extends EventEmitter implements Sayable {
     return rawMemberList.map(m => Contact.load(m.UserName))
   }
 
-  private parseNickMap(memberList: RoomRawMember[]): Map<string, string> {
-    const nickMap: Map<string, string> = new Map<string, string>()
-
+  /**
+   * ISSUE #64 emoji need to be striped
+   * ISSUE #104 never use remark name because sys group message will never use that
+   * @rui: cannot use argument NickName because it mix real nick and remark 
+   */
+  private parseMap(memberList: RoomRawMember[], parseContent: string): Map<string, string> {
```

Comment:
Add

`type NameType = 'nick' | 'display' | 'remark'`

Then change 

`parseContent: string`

to

`parseContent: NameType`